### PR TITLE
(#76) Added Dictionary Search and A-Z List component

### LINKED ISSUE #76: Enabler: Create the Search Box Display
# Create the search box display for the Term Page (without autosuggest).

> **ESTIMATE** 3

*Description of what is to be accomplished.*
* Add the "starts with" button
* Add the "contains" button
* Add a functional search box with search button
  * Clicking the search button should look at the input text and the "starts with" and "contains" radio buttons take me to the correct route for the search results (also will break, but the URL should be correct)
* Add the A-Z list
  * Clicking on the link should take me to the route for the expand of that letter (it will break, but the URL should be correct)

### Tasks
- [x] Create `Search` component ( #78 )
- [x] Create `A-Z` component ( #79 )

### LINKED ISSUE #78: Create Search component
*Create `Search` component.*

**Parent Issue:** #76

### Sub-Tasks
- [x] Add "Starts with" radio button (update component test)
- [x] Add "Contains" radio button (update component test)
- [x] Add searchbox and search button (update component test)
- [x] Add styling
- [x] Implement search button functionality to use provided form inputs to direct user to search route  (update component test)
- [x] Refactored `TextInput` component and added tests for `TextInput`, `Radio`, and `InputLabel` components

### LINKED ISSUE #79: Create A-Z component
Create A-Z component.*

**Parent Issue:** #76 

### Sub-Tasks
- [x] Create `A-Z` Component
- [x] Implement Unit Tests for Component

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #76
Closes #78 
Closes #79

### CODE REVIEW

File: src/components/az-list/AZList.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { useAppPaths } from '../../hooks/routing';
+
+const AZList = () => {
+    const { HomePath } = useAppPaths();
+    const baseURL = `${HomePath()}?expand=`;
+    const listArray = 'abcdefghijklmnopqrstuvwxyz#'.split('');
+    return (
+        <div className="list-container" data-testid={testIds.AZ_LIST}>
+            <div>
+                Browse:
```
Comment (arcepaul) [inline src/components/az-list/AZList.js]: Spanish text again.  `Ojear:`

**Comment by arcepaul** [comment]
@bryanpizzillo the answer is probably no but I'm guessing we're not going to use the Spanish alphabet for the spanish version?  I saw that it isn't the case for prod.

File: src/components/az-list/AZList.js (hunk 1 above)
Comment (seyilonge-nci) [reply src/components/az-list/AZList.js]: Updated for Spanish language

**Comment by bryanpizzillo** [comment]
> @bryanpizzillo the answer is probably no but I'm guessing we're not going to use the Spanish alphabet for the spanish version? I saw that it isn't the case for prod.

That is correct. We just have a-z and number for both. I don't know if I recall any terms that do not start with a-z/#.

File: src/components/az-list/AZList.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import { testIds } from "../../constants";
+import { useAppPaths } from "../../hooks/routing";
+import { useStateValue } from "../../store/store";
+import { i18n } from "../../utils/i18n";
+
+const AZList = () => {
+    const { HomePath } = useAppPaths();
+    const [{ language }] = useStateValue();
+    const baseURL = `${HomePath()}?expand=`;
```
Comment (bryanpizzillo) [inline src/components/az-list/AZList.js]: This should change to be the correct url. Will update in a minute.

Comment (seyilonge-nci) [reply src/components/az-list/AZList.js]: ```const baseURL = `${HomePath()}/expand=`;```?

Comment (bryanpizzillo) [reply src/components/az-list/AZList.js]: `useAppPaths` should be updated to contain:
`'Expand': '/expand/:expandChar'`
And then setting the URL should be done by calling `Expand(letter)`

Comment (seyilonge-nci) [reply src/components/az-list/AZList.js]: Got it! Thanks

File: src/components/az-list/AZList.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        <div className="list-container" data-testid={testIds.AZ_LIST}>
+            <div>
+                {i18n.browse[language]}:
+                <ul>
+                    { listArray.map( (item, i) => {
+                       const itemLink = item === '#' ? '%23' : item.toUpperCase();
+                       const label = item.toUpperCase();
+                       return (
+                           <li key={i} value={i+1}>
+                               <Link to={`${baseURL}${itemLink}`}>{label}</Link>
```
Comment (bryanpizzillo) [inline src/components/az-list/AZList.js]: `<Link to=` should be
`<Link to={Expand({ expandChar: itemLink})}`

File: src/components/az-list/AZList.js (hunk 2 above)
Comment (seyilonge-nci) [reply src/components/az-list/AZList.js]: Updated

File: src/components/az-list/AZList.js (hunk 3 above)
Comment (seyilonge-nci) [reply src/components/az-list/AZList.js]: Done!

File: src/components/az-list/AZList.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { useStateValue } from "../../store/store";
+import { i18n } from "../../utils/i18n";
+
+const AZList = () => {
+    const { ExpandPath } = useAppPaths();
+    const [{ language }] = useStateValue();
+    const listArray = 'abcdefghijklmnopqrstuvwxyz#'.split('');
+    return (
+        <div className="list-container" data-testid={testIds.AZ_LIST}>
+            <div>
```
Comment (arcepaul) [inline src/components/az-list/AZList.js]: replace this <div> with <nav>

Comment (seyilonge-nci) [reply src/components/az-list/AZList.js]: Done

**Pull request merged**